crowbar_framework/app/models/node_object.rb

```diff
@@ -203,6 +210,7 @@ def initialize(node)
         @role = NodeObject.create_new_role(node.name, node)
       else
         Rails.logger.fatal("Node exists without role!! #{node.name}")
+        raise raise Crowbar::Error::NotFound.new
```

Comment:
Wow, fixed. I wonder how it could even work in my tests... Ah, I guess I just pasted the changes I did on a real deployment to commit to git, that would explain it.